### PR TITLE
978: Catch exception on detect changes

### PR DESCRIPTION
**Pull request by zBritva**
Fix for [978](https://github.com/microsoft/charticulator/issues/978)

### CODE REVIEW

File: src/core/prototypes/state.ts [hunk 1]
```diff
@@ -153,12 +153,17 @@ export class ChartStateManager {
       for (let index = 0; index < origin.constraints.length; index++) {
         const originConstringts = origin.constraints[index];
```
Comment (MatthLee) [inline src/core/prototypes/state.ts:154]: nit: typo

File: src/core/prototypes/state.ts [hunk 2]
```diff
@@ -153,12 +153,17 @@ export class ChartStateManager {
...
         const current = chart.constraints[index];
-        expect_deep_approximately_equals(
-          originConstringts,
-          current,
-          defaultDifferenceApproximation,
-          true
-        );
+        try {
+          expect_deep_approximately_equals(
```
Comment (MatthLee) [inline src/core/prototypes/state.ts:157]: using throw/catch for deep equality check is not advised, but fixing this is outside the scope of this bug fix

File: src/core/prototypes/state.ts [hunk 3]
```diff
@@ -153,12 +153,17 @@ export class ChartStateManager {
...
+            originConstringts,
+            current,
+            defaultDifferenceApproximation,
+            true
+          );
+        } catch (ex) {
+          console.log(ex);
```
Comment (MatthLee) [inline src/core/prototypes/state.ts:164]: we probably don't need this console log, it just means the objects are different and we have changes to save, right?

**Review by MatthLee — APPROVED** [review]

File: src/core/prototypes/state.ts (hunk 3 above)
Comment (zBritva) [reply src/core/prototypes/state.ts:164]: Yes. It is

File: src/core/prototypes/state.ts (hunk 2 above)
Comment (zBritva) [reply src/core/prototypes/state.ts:157]: I will rework in next time

**Comment by zBritva** [comment]
@MatthLee  could you merge the PR?
